edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/config/BrokerConfigValidator.cs

```diff
@@ -58,7 +58,55 @@ public virtual IList<string> ValidateAuthorizationConfig(AuthorizationProperties
             return errors;
         }
 
-        private static void ValidateRule(AuthorizationProperties.Rule rule, int order, List<string> errors)
+        /// <summary>
+        /// Important!: Validation logic should be in sync with validation logic in the Broker.
+        ///
+        /// Validates bridge config and returns a list of errors (if any).
+        /// </summary>
+        public virtual IList<string> ValidateBridgeConfig(BridgeConfig properties)
+        {
+            Preconditions.CheckNotNull(properties, nameof(properties));
+
+            var order = 0;
+            var errors = new List<string>();
+            foreach (var bridge in properties)
+            {
+                ValidateBridge(bridge, order, errors);
+                order++;
+            }
+
+            return errors;
+        }
+
+        static void ValidateBridge(Bridge bridge, int order, List<string> errors)
+        {
+            if (string.IsNullOrEmpty(bridge.Endpoint))
+            {
+                errors.Add($"Bridge {order}: Endpoint must not be empty");
+            }
+
+            foreach (var setting in bridge.Settings)
+            {
```

Comment:
does the deserialization always resolve at least an empty list for bridge.Settings?

Reply:
good point. I marked some properties as required and added default values otherwise.